pkg/internal/block_device.go

```diff
@@ -39,17 +43,20 @@ type BlockDevice struct {
 	Children   []BlockDevice `json:"children,omitempty"`
 	Rotational string        `json:"rota"`
 	ReadOnly   string        `json:"ro,omitempty"`
-	PathByID   string        `json:"pathByID,omitempty"`
 	Serial     string        `json:"serial,omitempty"`
 	PartLabel  string        `json:"partLabel,omitempty"`
+
+	// DiskByPath is not part of lsblk output
+	// fetch and set it only if specified in the CR
+	DiskByPath string
 }
 
 // ListBlockDevices using the lsblk command
 func ListBlockDevices(exec Executor) ([]BlockDevice, error) {
 	// var output bytes.Buffer
 	var blockDeviceMap map[string][]BlockDevice
 	columns := "NAME,ROTA,TYPE,SIZE,MODEL,VENDOR,RO,STATE,KNAME,SERIAL,PARTLABEL,FSTYPE"
-	args := []string{"--json", "-o", columns}
+	args := []string{"--json", "--paths", "-o", columns}
```

Comment:
why is `--paths` argument needed?

Reply:
using --paths will give us the full path and make our job easy while comparing the paths and while adding paths into some commands we don't have to add `/dev/` anymore.